data-structures-queues/take-smaller.js

```diff
@@ -0,0 +1,25 @@
+/* exported takeSmaller */
+
+function takeSmaller(queue) {
+  const first = queue.dequeue();
+
+  while (queue.peek() !== undefined) {
```

Comment:
There doesn't need to be a loop here, it will always exit on the first iteration